# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: 5545fdc7573d
# Full Hash: 5545fdc7573d080566bf289b24f429fb941a6234
# Author: Robert Mader <robert.mader@posteo.de>
# Date: 2021-04-27 22:18:30
# Regressor Bug: 1707577
# File Overlap Count: 1
# Description:
#   Bug 1707577 - Assert on aInitData in Create() for popups, r=stransky
#   
#   These cases would have crashed. If these code paths actually can
#   get trigged, we'll now find out more easily.
#   
# ==============================================================================

diff -r 51929d701683 -r 5545fdc7573d widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Tue Apr 27 18:57:57 2021 +0000
+++ b/widget/gtk/nsWindow.cpp	Tue Apr 27 18:57:58 2021 +0000
@@ -4644,6 +4644,7 @@
       // popup window position.
       GtkWindowType type = GTK_WINDOW_TOPLEVEL;
       if (mWindowType == eWindowType_popup) {
+        MOZ_ASSERT(aInitData);
         type = (GdkIsX11Display() && aInitData->mNoAutoHide)
                    ? GTK_WINDOW_TOPLEVEL
                    : GTK_WINDOW_POPUP;
@@ -4724,6 +4725,7 @@
         }
 
       } else if (mWindowType == eWindowType_popup) {
+        MOZ_ASSERT(aInitData);
         mGtkWindowRoleName = "Popup";
 
         if (aInitData->mNoAutoHide) {
@@ -4894,6 +4896,7 @@
 #endif
 
       if (mWindowType == eWindowType_popup) {
+        MOZ_ASSERT(aInitData);
         // gdk does not automatically set the cursor for "temporary"
         // windows, which are what gtk uses for popups.
 
